docs/design/resource-strategy-fit-scheduling.md

```diff
@@ -0,0 +1,95 @@
+# Enhanced NodeResourceFit Plugin
+
+## Summary
+
+The NodeResourcesFit plug-in of native k8s can only adopt a type of strategy for all resources, such as MostRequestedPriority and LeastRequestedPriority. However, in industrial practice, this design does not apply to some scenarios. For example: In AI scenarios, businesses that apply for GPUs prefer to occupy the entire GPU machine first to prevent GPU fragmentation; businesses that apply for CPU & MEM are prioritized and dispersed to non-GPU machines to prevent excessive consumption of CPU & MEM on GPU machines, resulting in real tasks of applying for GPUs. Pending due to insufficient non-GPU resources
+. Therefore, two plugins are extended to solve this common problem.
+
+## Motivation
+
+- GPU tasks take priority over the entire GPU
+- CPU&MEM tasks are distributed to the CPU machine first
+
+## Design Consideration
+
+- The solution is more versatile, not limited to AI clusters or CPU clusters, and not limited to common CPU resources or extended GPU resources.
+
+- Different resource policies can be configured for different cluster types and prioritized in the form of weights.
+
+- Easy to expand
+
+### Goals
+
+- Different types of resources can be configured with different strategies to prioritize them in the form of weights
+
+- Prevent pods that have not applied for scarce resources from being scheduled to nodes with scarce resources.
+
+### Non-Goals
+
+- None.
+
+## Proposal
+
+Extend two plug-ins to meet the above needs
+
+- NodeResourcesFitPlus
+- ScarceResourceAvoidance
+
+## User Story
+
+### Story1
+- Users hope that different resource strategies can be adopted for different resource types. For example, in AI scenarios, they hope that pods that apply for GPU resources will occupy as many machines as possible, while pods that only apply for CPU resources will be evenly distributed to different machines.
+
+### Story2
+- Users hope that pods that have not applied for GPU resources should try not to schedule them on machines with GPU resources to prevent pending pods that really need GPU resources from being released due to insufficient CPU on the GPU machine.
+
+## Design Details
+
+### NodeResourcesFitPlus
+
+config：
+```
+resources: 
+  nvidia.com/gpu:
+    type: MostAllocated
+    weight: 2
+  cpu:
+    type: LeastAllocated
+    weight: 1
+  memory:
+    type: LeastAllocated
+    weight: 1
+```
+config description：
+
+<div align="center"><img width="582" height="393" src="images/resource_strategy_fit.png" /></div>
```

Comment:
![image](https://github.com/user-attachments/assets/0b8c7c52-f109-44c4-82a8-13085d318bc7)
There is an issue with the size of the image

Reply:
@JesseStutler I can't render the image locally. Can you give me some tips on how to adjust the parameters?

Reply:
done